lib/Dialect/Arc/Transforms/LowerClocksToFuncs.cpp

```diff
@@ -174,16 +185,15 @@ LogicalResult LowerClocksToFuncsPass::lowerClock(Operation *clockOp,
 /// body. Anything besides constants should no longer exist after a proper run
 /// of the pipeline.
 LogicalResult LowerClocksToFuncsPass::isolateClock(Operation *clockOp,
-                                                   Value modelStorageArg,
-                                                   Value clockStorageArg) {
+                                                   IRMapping storageMapping) {
   auto *clockRegion = &clockOp->getRegion(0);
   auto builder = OpBuilder::atBlockBegin(&clockRegion->front());
   DenseMap<Value, Value> copiedValues;
   auto result = clockRegion->walk([&](Operation *op) {
     for (auto &operand : op->getOpOperands()) {
       // Block arguments are okay, since there's nothing we can move.
-      if (operand.get() == modelStorageArg) {
-        operand.set(clockStorageArg);
+      if (storageMapping.contains(operand.get())) {
+        operand.set(storageMapping.lookup(operand.get()));
```

Comment:
Could you add a test to `test/Dialect/Arc/lower-clocks-to-funcs.mlir` that checks these new features. One or two new `arc.model` ops that trigger these additional arguments to be extracted would be nice. Could you then create a separate PR with just the improvements to LowerClocksToFuncs and that test?

Reply:
After looking closely, it turns out this code path is no longer used because we're not using two storage arguments in `arc.model`. Adding a test for this would trigger a verification error on `arc.model` saying it cannot have more than one argument, and I'm not confident to just drop that verification, and there is no longer a good reason to have more than one argument.

Therefore we reverted the changes in this file in the update.